### PR TITLE
Reduce cap by rent's leftover as temporary measure

### PR DESCRIPTION
**Pull request by ryoqun**
#### Problem

testnet's cap is busted now.

#### Summary of Changes

This is temporary fix, which can go without gating.

The underlying bug is now identified and the bug even resides in mainnet-beta/v1.2......

#### Context

found via capitalization work and blocking #11927

### CODE REVIEW

**Comment by ryoqun** [comment]
- [ ] reset capitalization on snapshot load at least on testnet to fix busted cap in snapshots....

File: runtime/src/bank.rs [hunk 1]
```diff
@@ -2099,7 +2100,9 @@ impl Bank {
             return;
         }
 
-        self.distribute_rent_to_validators(&self.vote_accounts(), rent_to_be_distributed);
+        let leftover =
+            self.distribute_rent_to_validators(&self.vote_accounts(), rent_to_be_distributed);
+        self.capitalization.fetch_sub(leftover, Ordering::Relaxed);
```
Comment (mvines) [inline runtime/src/bank.rs:2105]: Just to confirm, this is ok because `capitalization` is not a part of the bank hash right?

**Review by mvines — APPROVED** [review]
thanks!

File: runtime/src/bank.rs (hunk 1 above)
Comment (ryoqun) [reply runtime/src/bank.rs:2105]: > Just to confirm, this is ok because `capitalization` is not a part of the bank hash right?

Exactly. :)

**Comment by ryoqun** [comment]
>     * [x]  reset capitalization on snapshot load at least on testnet to fix busted cap in snapshots....

@mvines FYI, this is done. https://github.com/solana-labs/solana/pull/12111/commits/b00f1a51b87ee86a8545f71f065464a35115e1ea

**Pull request merged**